react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+
+export default class HotButton extends React.Component {
+  constructor(props) {
+    super(props);
+    const count = this.count = 0;
+    this.state = { clicks: count };
```

Comment:
With a few exceptions, we want to avoid attaching any additional information to the `this` object. The information can/should almost always be stored in state.